Mobile/ShopLens/Droid/SpeechActivity.cs

```diff
@@ -49,36 +61,78 @@ protected override void OnCreate(Bundle savedInstanceState)
             {
                 recButton.Click += (sender, e) =>
                 {
-                    // change the text on the button
-                    recButton.Text = "End Recording";
-                    isRecording = !isRecording;
-                    if (isRecording)
-                    {
-                        // create the intent and start the activity
-                        var voiceIntent = new Intent(RecognizerIntent.ActionRecognizeSpeech);
-                        voiceIntent.PutExtra(RecognizerIntent.ExtraLanguageModel, RecognizerIntent.LanguageModelFreeForm);
-
-                        // put a message on the modal dialog
-                        voiceIntent.PutExtra(RecognizerIntent.ExtraPrompt, Application.Context.GetString(Resource.String.messageSpeakNow));
-
-                        // if there is more then 1.5s of silence, consider the speech over
-                        voiceIntent.PutExtra(RecognizerIntent.ExtraSpeechInputCompleteSilenceLengthMillis, 1500);
-                        voiceIntent.PutExtra(RecognizerIntent.ExtraSpeechInputPossiblyCompleteSilenceLengthMillis, 1500);
-                        voiceIntent.PutExtra(RecognizerIntent.ExtraSpeechInputMinimumLengthMillis, 15000);
-                        voiceIntent.PutExtra(RecognizerIntent.ExtraMaxResults, 1);
-
-                        // you can specify other languages recognised here, for example
-                        // voiceIntent.PutExtra(RecognizerIntent.ExtraLanguage, Java.Util.Locale.German);
-                        // if you wish it to recognise the default Locale language and German
-                        // if you do use another locale, regional dialects may not be recognised very well
-
-                        voiceIntent.PutExtra(RecognizerIntent.ExtraLanguage, Java.Util.Locale.Default);
-                        StartActivityForResult(voiceIntent, VOICE);
-                    }
+                    DisableVoicer();
+                    RecordVoice(recButton);
                 };
+
+                recAndVoiceButton.Click += (sender, e) =>
+                {
+                    EnableVoicer();
+                    RecordVoice(recAndVoiceButton); 
+                };
+            }
+        }
+
+        public void OnInit([GeneratedEnum] OperationResult status)
```

Comment:
More about this method can be found here:
https://developer.xamarin.com/api/member/Android.Speech.Tts.TextToSpeech+IOnInitListener.OnInit/p/Android.Speech.Tts.OperationResult/